packages/documentsearch-extension/src/searchoverlay.tsx

```diff
@@ -0,0 +1,285 @@
+const CLOSE_BUTTON_CLASS = 'jp-DocumentSearch-close-button';
+const REGEX_ERROR_CLASS = 'jp-DocumentSearch-regex-error';
+
+interface ISearchEntryProps {
+  onCaseSensitiveToggled: Function;
+  onRegexToggled: Function;
+  onKeydown: Function;
+  onChange: Function;
+  caseSensitive: boolean;
+  useRegex: boolean;
+  inputText: string;
+  forceFocus: boolean;
+}
+
+class SearchEntry extends React.Component<ISearchEntryProps> {
+  constructor(props: ISearchEntryProps) {
+    super(props);
+  }
+
+  componentDidMount() {
+    this.focusInput();
+  }
+
+  focusInput() {
+    (this.refs.searchInputNode as HTMLInputElement).focus();
+  }
+
+  componentDidUpdate() {
+    if (this.props.forceFocus) {
+      this.focusInput();
+    }
+  }
+
+  render() {
+    const caseButtonToggleClass = this.props.caseSensitive
+      ? CASE_BUTTON_CLASS_ON
+      : CASE_BUTTON_CLASS_OFF;
+    const regexButtonToggleClass = this.props.useRegex
+      ? REGEX_BUTTON_CLASS_ON
+      : REGEX_BUTTON_CLASS_OFF;
+
+    return (
+      <div className={INPUT_WRAPPER_CLASS}>
+        <input
+          placeholder={this.props.inputText ? null : 'SEARCH'}
+          className={INPUT_CLASS}
+          value={this.props.inputText}
+          onChange={e => this.props.onChange(e)}
+          onKeyDown={e => this.props.onKeydown(e)}
+          ref="searchInputNode"
+        />
+        <button
+          className={caseButtonToggleClass}
+          onClick={() => this.props.onCaseSensitiveToggled()}
+        />
+        <button
+          className={regexButtonToggleClass}
+          onClick={() => this.props.onRegexToggled()}
+        />
+      </div>
+    );
+  }
+}
+
+interface IUpDownProps {
+  onHighlightPrevious: Function;
+  onHightlightNext: Function;
+}
+
+function UpDownButtons(props: IUpDownProps) {
+  return (
+    <div className={UP_DOWN_BUTTON_WRAPPER_CLASS}>
+      <button
+        className={UP_BUTTON_CLASS}
+        onClick={() => props.onHighlightPrevious()}
+      />
+      <button
+        className={DOWN_BUTTON_CLASS}
+        onClick={() => props.onHightlightNext()}
+      />
+    </div>
+  );
+}
+
+interface ISearchIndexProps {
+  currentIndex: number;
+  totalMatches: number;
+}
+
+function SearchIndices(props: ISearchIndexProps) {
+  return (
+    <>
+      <div className={INDEX_COUNTER_CLASS}>
+        {props.totalMatches === 0
+          ? '-/-'
+          : `${props.currentIndex + 1}/${props.totalMatches}`}
+      </div>
+    </>
+  );
+}
+
+interface ISearchOverlayProps {
+  overlayState: IDisplayState;
+  onCaseSensitiveToggled: Function;
+  onRegexToggled: Function;
+  onHightlightNext: Function;
+  onHighlightPrevious: Function;
+  onStartSearch: Function;
+  onEndSearch: Function;
+}
+
+class SearchOverlay extends React.Component<
+  ISearchOverlayProps,
+  IDisplayState
+> {
+  constructor(props: ISearchOverlayProps) {
+    super(props);
+    this.state = props.overlayState;
+  }
+
+  private onChange(event: React.ChangeEvent) {
```

Comment:
(for here and elsewhere): our convention is that private variables and methods start with an underscore, like `_onChange`.

Reply:
(this and other conventions are noted in the style guide: https://github.com/jupyterlab/jupyterlab/wiki/TypeScript-Style-Guide)